torch/utils/checkpoint.py

```diff
@@ -104,17 +127,20 @@ def backward(ctx, *args):
         # run backward() with only tensor that requires grad
         outputs_with_grad = []
         args_with_grad = []
+        tensor_args_index = 0
         for i in range(len(outputs)):
-            if outputs[i].requires_grad:
+            if torch.is_tensor(outputs[i]) and outputs[i].requires_grad:
                 outputs_with_grad.append(outputs[i])
-                args_with_grad.append(args[i])
+                args_with_grad.append(args[tensor_args_index])
```

Comment:
This indexing feels bad. Following the update to include None in the grad_output, that will simplify this code a lot!